code/modules/research/bepis.dm

```diff
@@ -0,0 +1,248 @@
+	for(var/obj/item/stock_parts/scanning_module/Scan in component_parts)
+		S += ((Scan.rating - 1) * 0.25)
+	inaccuracy_percentage = (1.5 - S)
+
+/obj/machinery/rnd/bepis/proc/depositcash()
+	var/deposit_value = 0
+	if(!Card)
+		say("No account detected.")  //No freeloading off of science.
+		return
+	else if(Card.registered_account)
+		account = Card.registered_account
+
+	deposit_value = banking_amount
+	if(deposit_value == null || deposit_value == 0)
+		chamber_status = 1
+		update_icon_state()
+		say("Attempting to deposit 0 credits. Aborting.")
+		return
+	deposit_value = CLAMP(round(deposit_value, 1), 1, 30000)
+
+	if(!account.has_money(deposit_value))
+		say("You do not possess enough credits.")
+		return
+	else
+		account.adjust_money(-deposit_value) //The money vanishes, not paid to any accounts.
+		banked_cash += deposit_value
+		use_power(1000 * power_saver)
+		say("Cash deposit successful. There is [banked_cash] in the chamber.")
+
+	if(banked_cash >= 1)
+		chamber_status = 2
+		update_icon_state()
+	else
+		chamber_status = 1
+		update_icon_state()
+	return
+
+/obj/machinery/rnd/bepis/proc/calcsuccess()
+	var/gauss_major = 0
+	var/gauss_minor = 0
+	var/gauss_real = 0
+	gauss_major = (gaussian(major_threshold, std) - negative_cash_offset)	//This is the randomized profit value that this experiment has to surpass to unlock a tech.
+	gauss_minor = (gaussian(minor_threshold, std) - negative_cash_offset)	//And this is the threshold to instead get a minor prize.
+	gauss_real = (gaussian(banked_cash, std*inaccuracy_percentage) + positive_cash_offset)	//this is the randomized profit value that your experiment expects to give.
+	say("Real, [gauss_real]. Minor, [gauss_minor]. Major, [gauss_major].")
+	if(gauss_real >= gauss_major) //Major Success.
+		success_type = 2
+		return
+	else if(gauss_real >= gauss_minor) //Minor Success.
+		success_type = 1
+		return
+	else if(gauss_real <= -1)
+		success_type = 3 //Critical Failure.
+		return
+	success_type = 0 //Neutral Failure.
+	return
+
+/obj/machinery/rnd/bepis/update_icon_state()
+	if(is_operational() && (chamber_status < 2))
+		chamber_status = 1
+	if(((powered == FALSE) && (banked_cash == 0)) || (!is_operational()))
+		chamber_status = 0
+	if((powered == TRUE) && (banked_cash > 0))
+		chamber_status = 2
+	else if ((powered == FALSE) && (banked_cash > 0))
+		chamber_status = 3
+	switch(chamber_status)
+		if(0)
+			icon_state = "chamber"
+		if(1)
+			icon_state = "chamber_active"
+		if(2)
+			icon_state = "chamber_active_loaded"
+		if(3)
+			icon_state = "chamber_loaded"
+
+
+/obj/machinery/rnd/bepis/ui_interact(mob/user, ui_key = "main", datum/tgui/ui = null, force_open = 0, datum/tgui/master_ui = null, datum/ui_state/state = GLOB.default_state)
+	ui = SStgui.try_update_ui(user, src, ui_key, ui, force_open)
+	if(!ui)
+		ui = new(user, src, ui_key, "bepis", name, 500, 400, master_ui, state)
+		ui.open()
+	if(ishuman(user))
+		H = user
+		Card = H.get_idcard(TRUE)
+		update_icon_state()
```

Comment:
update_icon doesn't seem to depend on either of these variables, you're calling it unconditionally later and these two seem unnecessary,